duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/inputscreen/ui/InputScreenFragment.kt

```diff
@@ -282,6 +313,56 @@ class InputScreenFragment : DuckDuckGoFragment(R.layout.fragment_input_screen) {
         }.launchIn(lifecycleScope)
     }
 
+    private fun configureLogoAnimation() {
+        if (appTheme.isLightModeEnabled()) {
+            binding.ddgLogo.setAnimation(R.raw.duckduckgo_ai_transition_light)
+        } else {
+            binding.ddgLogo.setAnimation(R.raw.duckduckgo_ai_transition_dark)
+        }
+    }
+
+    private fun updateLogoAnimationForScroll(position: Int, positionOffset: Float) {
+        binding.ddgLogo.apply {
+            setMinAndMaxFrame(0, 15)
+            if (viewModel.hasNewTabContent()) {
```

Comment:
```suggestion
            if (!viewModel.visibilityState.value.showSearchLogo) {
```
I think this is closer to what we need. Also, if applied with the suggestion above, we could avoid exposing `viewModel.hasNewTabContent()` altogether.

Reply:
I was doing that before but it was causing that animation glitch you found, so I’m checking `hasNewTabContent` instead. It’s less of an issue now since all the logic has been moved to the ViewModel.